capten/agent/pkg/agent/plugin_argocd_apis.go

```diff
@@ -185,3 +185,15 @@ func (a *Agent) deleteProjectFromArgoCD(ctx context.Context, projectUrl string)
 	}
 	return nil
 }
+
+func (a *Agent) isProjectRegisteredWithArgoCD(ctx context.Context, projectUrl string) (bool, error) {
+	argocdClient, err := argocd.NewClient(&logging.Logging{})
+	if err != nil {
+		return false, err
+	}
+	_, err = argocdClient.GetRepository(ctx, projectUrl)
+	if err != nil {
+		return false, err
```

Comment:
err could happen with  connectivity or other cases.
check err that repo not exist

Reply:
ok